### ISSUE TITLE
Test.Neon.Cadence executable (cadence-proxy.exe) incorrect format for this OS

### ISSUE THREAD

**Issue by johncburns1** [nforgeio/neonKUBE#505]
The cadence-proxy.exe Golang executable that stored for testing the cadence-proxy in Test.Neon.Cadence/bin/Debug/netcoreapp3.0 is in the wrong format to execute on the Windows OS.  When I run the tests, I get the exception:
```
Message: System.NullReferenceException : Object reference not set to an instance of an object.
```

Under the surface, this is because the executable cannot be started.  The error is:

```
"The specified executable is not a valid application for this OS platform." System.ComponentModel.Win32Exception
```

This error shows up after I rebuild and then run Test/Test.Neon.Cadence/Test_Messages.cs with 

```
DebugPrelaunched       = false,
DebugDisableHandshakes = false,
DebugDisableHeartbeats = true,
```

**Comment by johncburns1** [comment]
Upon further debugging, this is an issue that persists even when I manually replace the file, and size shrinks to 0KB as seen in the file explorer.

**Comment by johncburns1** [comment]
Stack Trace: 
Test Name:	TestCadence.Test_Messages.TestConnectRequest
Test FullName:	TestCadence.Test_Messages.TestConnectRequest
Test Source:	C:\Users\johnc\source\repos\nforgeio\neonKUBE\Test\Test.Neon.Cadence\Test_Messages.cs : line 645
Test Outcome:	Failed
Test Duration:	0:00:00.001

Result StackTrace:	
at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Neon.Cadence.CadenceConnection.StartProxy(IPEndPoint endpoint, CadenceSettings settings) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Cadence\CadenceConnection.cs:line 347
   at Neon.Cadence.CadenceConnection..ctor(CadenceSettings settings) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Cadence\CadenceConnection.cs:line 439
   at Neon.Xunit.Cadence.CadenceFixture.StartAsComposed(CadenceSettings settings, String image, String name, String[] env, Boolean emulateProxy) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Xunit.Cadence\CadenceFixture.cs:line 167
   at Neon.Xunit.Cadence.CadenceFixture.<>c__DisplayClass4_0.<Start>b__0() in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Xunit.Cadence\CadenceFixture.cs:line 106
   at Neon.Xunit.TestFixture.Start(Action action) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Xunit\Fixtures\TestFixture.cs:line 231
   at Neon.Xunit.Cadence.CadenceFixture.Start(CadenceSettings settings, String image, String name, String[] env, Boolean emulateProxy) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Lib\Neon.Xunit.Cadence\CadenceFixture.cs:line 103
   at TestCadence.Test_Messages..ctor(CadenceFixture fixture) in C:\Users\johnc\source\repos\nforgeio\neonKUBE\Test\Test.Neon.Cadence\Test_Messages.cs:line 62
Result Message:	System.ComponentModel.Win32Exception : The specified executable is not a valid application for this OS platform.

**Comment by johncburns1** [comment]
Update: The Windows exe was not being completely built by the solution build, so the executable was just an empty .exe which is why it could not be executed.  I ran the Go/build-all.ps1 powershell script to debug, and the resulting output was:
```
go build command-line-arguments: build output "C:\\Users\\johnc\\source\\repos\\nforgeio\\neonKUBE\\Build\\cadence-proxy.win.exe" already exists and is not an object file
C:\Users\johnc\source\repos\nforgeio\neonKUBE\Go\build-cadence-proxy.ps1 : *** ERROR: [cadence-proxy] WINDOWS build failed.  Check build logs: C:\Users\johnc\source\repos\nforgeio\neonKUBE\Build\build-cadence-proxy.log
At C:\Users\johnc\source\repos\nforgeio\neonKUBE\Go\build-all.ps1:39 char:1
+ ./build-cadence-proxy.ps1 -buildConfig $buildConfig
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,build-cadence-proxy.ps1
```

I believe that this may have been an issue of a build failure because of errors in the Golang cadence-proxy when trying to debug the WinDesktop with Marcus.  I deleted the files in build and rebuilt and the exe's seem to be in proper form to use now.

**Comment by johncburns1** [comment]
The issue was not resolved.  The cadence-proxy.exe file is overridden every time in the Test/Test.Neon.Cadence/Debug/netcoreapp3.0 dir to an exe that is not usable.

**Comment by jefflill** [comment]
It looks like this was caused by commit 135b0252442900ae0ff0de3fb2dec4211a035dab you made Tuesday afternoon.  I'm betting that you had to resolve a merge conflict for `Neon.Cadence.csproj` and we lost the settings that actually imported the cadence-proxy GZ files into the assembly.  Here's how the file diff looks:

![image](https://user-images.githubusercontent.com/4908431/57081726-315e1400-6caa-11e9-9224-5cc8ad65cee1.png)

I corrected this by replacing the `<Folder Include="Resources\" />` line with the `<EmbeddedResource.../>` settings and I also added an explicit check that verifies that the resource actually exists and throws a nicer exception when it doesn't.

This explains why this worked on the MASTER branch but not CADENCE; because the change hasn't been merged to MASTER yet.

Merge conflicts are a bitch.  The good news is that our unit tests actually caught this problem.
